monix-execution/shared/src/main/scala/monix/execution/misc/Local.scala

```diff
@@ -243,7 +182,23 @@ object Local {
       r
     }
 
-    final def mkIsolated: Context = {
+    final def bind(key: Key, value: Option[Any]): Context =
+      new Bound(key, value.orNull, value.isDefined, this)
+
+    final def isolate(): Context =
+      isolateLoop()
+
+    /**
+      * DEPRECATED — renamed to [[isolate]].
+      */
+    @deprecated("Renamed to isolate()", "3.0.0")
+    private[misc] def mkIsolated(): Unbound = {
+      // $COVERAGE-OFF$
+      isolateLoop()
+      // $COVERAGE-ON$
+    }
+
+    private[this] final def isolateLoop(): Unbound =
```

Comment:
Why move the implementation out?

Reply:
To reuse it in both the deprecated method and the new one, as they are the same, only the return type is different.

Reply:
Ah, didn't notice difference in return types